### DISCUSSION TITLE
The Android App Bundle was signed with the wrong key.

### DISCUSSION THREAD

**Discussion by eduardoazvd17** [codemagic-ci-cd/codemagic-docs#2808] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Upload code sign file .keystore or .jks no one works.

### Expected results

Generate android build using the code sign file i've uploaded.

### Actual results

This is the last one log, but every time i try the value from "Found" changes, then i conclude that codemagic is not using my attached file... I already checked my file SHA-1 and its fine, its correct.

```
Publishing failed :|
Google Play failed to upload artefacts. The Android App Bundle was signed with the wrong key. Found: SHA1: B9:55:...:91:2B, expected: SHA1: 7D:31:...:03:23.: {
    "error": {
        "code": 403,
        "message": "The Android App Bundle was signed with the wrong key. Found: SHA1: B9:55:...:91:2B, expected: SHA1: 7D:31:...:03:23.",
        "status": "PERMISSION_DENIED"
    }
}


```

### Build id (optional)

6722d3fef5efaf48a4e0eb89

**Comment by dam-ease** [comment]
Hello @eduardoazvd17,
The error indicates that the uploaded binary was signed with a keystore that differs from the one currently in use. You’ll need to locate the original keystore with the expected SHA signature: SHA1: 7D:31:...:03:23.

Please can you check the SHA signature of your keystore, using this command in your terminal:
```
keytool -list -v -keystore your_keystore_file.keystore
```
If the original keystore is lost, you can request a key change through Google support page [here](https://support.google.com/googleplay/android-developer/contact/key).

**Comment by eduardoazvd17** [comment]
How i said, i already checked and its the same as the currently in use on play store... but i already solved, it was my fault, i forget to change on `build.gradle` to use the `key.properties` file.

Thanks for reply, have a nice day.

**Reply by dam-ease** [reply]
Alright!
Glad to hear, happy building!
